src/index.ts

```diff
@@ -381,6 +381,13 @@ export function match<P extends object = object>(
   return regexpToFunction<P>(re, keys, options);
 }
 
+export interface RegExpWithKeys extends RegExp {
+  /**
+   * Optional property that includes key metadata corresponding to this regex.
+   */
+  keys?: Key[];
```

Comment:
Can we make this required? A way to get around TypeScript complaining is to use `Object.assign(regex, { keys })`.